cast/connection.go

```diff
@@ -24,7 +24,16 @@ const (
 	dialerKeepAlive = time.Second * 30
 )
 
-type Connection struct {
+type Connection interface {
+	Start(addr string, port int) error
+	MsgChan() chan *pb.CastMessage
+	Close() error
+	SetDebug(debug bool)
+	LocalAddr() (addr string, err error)
+	Send(requestID int, payload Payload, sourceID, destinationID, namespace string) error
+}
+
+type connection struct {
```

Comment:
The same with this, it is good practise to keep these public if access to the structs are needed.